ci/base-tests.sh

```diff
@@ -29,26 +33,27 @@ export CARGO_TARGET_DIR=`pwd`/target/
 (
   # Check sysroot handling
   sysroot=$(./target/debug/clippy-driver --print sysroot)
-  test $sysroot = $(rustc --print sysroot)
+  test "$sysroot" = "$(rustc --print sysroot)"
 
-  if [ -z $OS_WINDOWS ]; then
+  if [[ -z "$OS_WINDOWS" ]]; then
     desired_sysroot=/tmp
   else
     desired_sysroot=C:/tmp
   fi
   sysroot=$(./target/debug/clippy-driver --sysroot $desired_sysroot --print sysroot)
-  test $sysroot = $desired_sysroot
+  test "$sysroot" = $desired_sysroot
 
   sysroot=$(SYSROOT=$desired_sysroot ./target/debug/clippy-driver --print sysroot)
-  test $sysroot = $desired_sysroot
+  test "$sysroot" = $desired_sysroot
 
   # Make sure this isn't set - clippy-driver should cope without it
   unset CARGO_MANIFEST_DIR
 
   # Run a lint and make sure it produces the expected output. It's also expected to exit with code 1
-  # XXX How to match the clippy invocation in compile-test.rs?
-  ! ./target/debug/clippy-driver -Dwarnings -Aunused -Zui-testing --emit metadata --crate-type bin tests/ui/cstring.rs 2> cstring.stderr
-  diff <(sed -e 's,tests/ui,$DIR,' -e '/= help/d' cstring.stderr) tests/ui/cstring.stderr
+  # FIXME: How to match the clippy invocation in compile-test.rs?
+  ./target/debug/clippy-driver -Dwarnings -Aunused -Zui-testing --emit metadata --crate-type bin tests/ui/cstring.rs 2> cstring.stderr || true
```

Comment:
I think the `! _` is the better solution for a command that _should_ fail.

Reply:
```suggestion
  test ! ./target/debug/clippy-driver -Dwarnings -Aunused -Zui-testing --emit metadata --crate-type bin tests/ui/cstring.rs 2> cstring.stderr
```
would be better I think?